core/chains/evm/logpoller/log_poller.go

```diff
@@ -1096,3 +1104,69 @@ func EvmWord(i uint64) common.Hash {
 	binary.BigEndian.PutUint64(b, i)
 	return common.BytesToHash(b)
 }
+
+type notifyEntry struct {
+	index    int
+	value    common.Hash
+	notifyCh chan struct{}
+}
+
+// Notify returns a channel that emits events when a new log with the given topic key and value is saved.
+func (lp *logPoller) Notify(topicIndex int, topicValue common.Hash) <-chan struct{} {
+	if topicIndex < 0 {
+		panic("cannot have negative topic index")
```

Comment:
check `validateTopicIndex` in logpoller/orm.go

Reply:
I made this change, but now I'm running into problems with unit tests. Previously I used topic index = 0 in unit tests, but 0 is not valid according to validateTopicIndex. The problem is that almost no logs in tests have more than 1 topic, and I'm not familiar enough with the overall stack to know how to generate some test logs with >1 topics. I tried to follow the code, but I couldn't understand the interaction between the test harness and the log poller dependencies.

Do you have any pointers for how I can effectively test this?